### PR TITLE
Load genes from pre-downloaded file

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Partially fixing #104

**How to prepare for test**:
- Download genes files (either build 37 or 38) using [schug](https://github.com/Clinical-Genomics/schug)

### How to test:
- Launch a server and use the modified load genes andpoint to load the genes using the pre-downloaded file

### Expected outcome:
- [x] Genes should be loaded

### Review:
- [ ] Code approved by
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by northwestwitch** [comment]
# Tested locally, given a pre-downloaded genes (build 38) file:

```
(schug) chiararasi@n177-p182 genes_etc % pwd
/Users/chiararasi/Documents/work/genes_etc
(schug) chiararasi@n177-p182 genes_etc % head -n 15 genes_GRCh38.txt
Chromosome/scaffold name	Gene start (bp)	Gene end (bp)	Gene stable ID	HGNC symbol	HGNC ID
1	1471765	1497848	ENSG00000160072	ATAD3B	HGNC:24007
1	182696	184174	ENSG00000279928	DDX11L17	HGNC:55080
1	2581560	2584533	ENSG00000228037
1	3069168	3438621	ENSG00000142611	PRDM16	HGNC:14000
1	5301928	5307394	ENSG00000284616
1	2403964	2413797	ENSG00000157911	PEX10	HGNC:8851
1	2350414	2352820	ENSG00000269896
1	257864	359681	ENSG00000228463
1	5492978	5494674	ENSG00000260972
1	10054445	10054781	ENSG00000224340
1	3944547	3949024	ENSG00000226374	LINC01345	HGNC:50564
1	4175528	4175899	ENSG00000229280	EEF1DP6	HGNC:51456
1	10472288	10630758	ENSG00000142655	PEX14	HGNC:8856
1	4571481	4594016	ENSG00000232596	LINC01646	HGNC:52433
```

Load genes from file:

image

File: src/chanjo2/meta/handle_bed.py [hunk 1]
```diff
@@ -1,4 +1,11 @@
-from typing import List, Tuple
+from typing import List, Tuple, Iterator
+
+
+def resource_lines(file_path: str) -> Iterator[str]:
```
Comment (northwestwitch) [inline src/chanjo2/meta/handle_bed.py:4]: This was moved here from src/chanjo2/populate_demo.py

File: src/chanjo2/populate_demo.py [hunk 2]
```diff
@@ -46,13 +47,6 @@
 ]
 
 
-def resource_lines(file_path: str) -> Iterator[str]:
```
Comment (northwestwitch) [inline src/chanjo2/populate_demo.py:49]: Moving it to src/chanjow/meta/handle_bed, so it's used more generally from the whole app

**Comment by northwestwitch** [comment]
Hi @henrikstranneheim, when you have time I could use a review for this PR. This is the first of 3. The other 2 will allow loading of transcripts and exons as well from pre-downloaded files!

File: tests/src/chanjo2/endpoints/test_intervals.py [hunk 3]
```diff
@@ -61,6 +61,37 @@ def test_load_genes(
...
+    assert response.status_code == status.HTTP_200_OK
+
+    # THEN all the genes should be loaded
+    assert response.json()["detail"] == f"{nr_genes} genes loaded into the database"
+
+    # WHEN sending a request to the "genes" endpoint
+    response: Response = client.post(endpoints.GENES, json={"build": build})
+    assert response.status_code == status.HTTP_200_OK
+    result = response.json()
+    # THEN the expected number of genes should be returned
```
Comment (henrikstranneheim) [inline tests/src/chanjo2/endpoints/test_intervals.py:89]:
```suggestion
    result = response.json()
    
    # THEN the expected number of genes should be returned
```

File: tests/src/chanjo2/endpoints/test_intervals.py [hunk 4]
```diff
@@ -61,6 +61,37 @@ def test_load_genes(
...
+    assert len(result) == nr_genes
+    # AND the gene should have the right format
+    assert Gene(**result[0])
```
Comment (henrikstranneheim) [inline tests/src/chanjo2/endpoints/test_intervals.py:92]:
```suggestion

    # THEN the gene should have the right format
    assert Gene(**result[0])
```

**Review by henrikstranneheim — APPROVED** [review]
Looks great 👍

**Comment by northwestwitch** [comment]
Looks like black didn't like the extra space. sorry! Thanks for the review @henrikstranneheim !

**Pull request merged**